src/Services/src/Serialization/SerializationService.cs

```diff
@@ -22,12 +25,40 @@ public SerializationService()
 
         public AppManifest ParseCfAppManifest(string manifestContents)
         {
-            return _cfAppManifestParser.Deserialize<AppManifest>(manifestContents);
+            var manifest = _cfAppManifestParser.Deserialize<AppManifest>(manifestContents);
+            var manifestApp = manifest.App;
+
+            if (manifestApp.Buildpacks == null && manifestApp.Buildpack != null)
+            {
+                manifest.OriginalBuildpackScheme = AppManifest.BuildpackScheme.Singular;
+                ReformatSingularBuildpackAsList(manifest, manifestApp.Buildpack);
+            }
+
+            return manifest;
         }
 
         public string SerializeCfAppManifest(AppManifest manifest)
         {
+            var manifestApp = manifest.App;
+            if (manifestApp.Buildpacks != null && manifestApp.Buildpacks.Count == 1 && manifest.OriginalBuildpackScheme == AppManifest.BuildpackScheme.Singular)
```

Comment:
Consider breaking this up into multiple lines.  
```
if (manifestApp.Buildpacks != null 
    && manifestApp.Buildpacks.Count == 1 
    && manifest.OriginalBuildpackScheme == AppManifest.BuildpackScheme.Singular)
```

Reply:
Changed in b05610d